CONTRIBUTING.md

```diff
@@ -0,0 +1,40 @@
+# Contributing
+
+## Reporting Issues
+
+If you have found what you think is a bug, please [file an issue](https://github.com/pmndrs/jotai/issues/new).
+
+For usage questions, prefer [starting a discussion](https://github.com/pmndrs/jotai/discussions/new).
+
+## Suggesting new features
+
+If you are here to suggest a feature, first create an issue if it does not already exist. From there, we will discuss use-cases for the feature and then finally discuss how it could be implemented.
+
+## Development
+
+If you have been assigned to fix an issue or develop a new feature, please follow these steps to get started:
+
+- Fork this repository.
+- Install dependencies by running `$ yarn`.
+- Implement your changes and run test files or create new test files.
+- If you want to test it in a React project you can either use `$ yarn link` or `yalc` package.
+- Git stage your required changes and commit (see below commit guidelines).
+- Submit PR for review.
```

Comment:
So, is this your recommended workflow for development?

Reply:
As someone non-initiated into the repo, this is roughly similar to what I did to contribute. I ended up writing a failing test, then actively working on in running `yarn test:dev` and once I was ready to test it in `jotai`, I used `yalc` to test it locally.

So I'd say that from my own perspective @TheEhsanSarshar is rather close here! 

Also I gotta say, as a comment to the overall necessity of CONTRIBUTING.md, I did indeed spend a few minutes looking for it and it was a little annoying having to "reverse-engineer" how to contribute. So I think it's definitely a plus having this in a repo.

Reply:
@israelidanny Great feedback. TBH, I wasn't confident how this document would help contributors.
So, based on your reverse-engineered experience, would you review the entire file and give suggestions?

Reply:
dear @israelidanny please do it.

Reply:
As far as the list, I would perhaps suggest changing `Implement your changes and run test files or create new test files.` and split it to to something like:
-  Create failing tests for your fix or new feature
-  Implement your changes and confirm that all test are passing. You can run the tests continuously during development via the `yarn test:dev` command

IMHO this would encourage more TDD and keep the contributions' test coverage and quality up.

Reply:
I personally follow this workflow.